internal/pkg/agent/application/upgrade/artifact/download/localremote/downloader.go

```diff
@@ -21,9 +21,12 @@ func NewDownloader(log *logger.Logger, config *artifact.Config) (download.Downlo
 	downloaders := make([]download.Downloader, 0, 3)
 	downloaders = append(downloaders, fs.NewDownloader(config))
 
+	// if the current build is a snapshot we use this downloader to update
+	// to the latest snapshot of the same version useful for testing with
+	// a snapshot version of fleet for example
```

Comment:
Thanks for adding this comment.  
```suggestion
	// If the current build is a snapshot we use this downloader to update
	// to the latest snapshot of the same version. Useful for testing with
	// a snapshot version of fleet, for example.
```